src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,587 @@
+        /// </returns>
+        /// <remarks>
+        ///   <b>Matches</b> is used to ensure data integrity.
+        /// </remarks>
+        public bool Matches(Stream data)
+        {
+            var digest = Algorithm.Hasher().ComputeHash(data);
+            for (var i = digest.Length - 1; 0 <= i; --i)
+                if (digest[i] != Digest[i])
+                    return false;
+
+            return true;
+        }
+
+        private void RaiseUnknownHashingAlgorithm(HashingAlgorithm algorithm)
+        {
+            if (Log.IsWarnEnabled)
+                Log.WarnFormat("Unknown hashing algorithm number 0x{0:x2}.", algorithm.Code);
+
+            var handler = UnknownHashingAlgorithm;
+            if (handler != null)
+            {
+                var args = new UnknownHashingAlgorithmEventArgs
+                {
+                    Algorithm = algorithm
+                };
+                handler(this, args);
+            }
+        }
+
+        /// <summary>
+        ///   Generate the multihash for the specified byte array. 
+        /// </summary>
+        /// <param name="data">
+        ///   The byte array containing the data to hash.
+        /// </param>
+        /// <param name="algorithmName">
+        ///   The name of the hashing algorithm to use; defaults to <see cref="DefaultAlgorithmName"/>.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="MultiHash"/> for the <paramref name="data"/>.
+        /// </returns>
+        public static MultiHash ComputeHash(byte[] data, string algorithmName = DefaultAlgorithmName)
+        {
+            using (var alg = GetHashAlgorithm(algorithmName))
+            {
+                return new MultiHash(algorithmName, alg.ComputeHash(data));
+            }
+        }
+
+        /// <summary>
+        ///   Generate the multihash for the specified <see cref="Stream"/>. 
+        /// </summary>
+        /// <param name="data">
+        ///   The <see cref="Stream"/> containing the data to hash.
+        /// </param>
+        /// <param name="algorithmName">
+        ///   The name of the hashing algorithm to use; defaults to <see cref="DefaultAlgorithmName"/>.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="MultiHash"/> for the <paramref name="data"/>.
+        /// </returns>
+        public static MultiHash ComputeHash(Stream data, string algorithmName = DefaultAlgorithmName)
+        {
+            using (var alg = GetHashAlgorithm(algorithmName))
+            {
+                return new MultiHash(algorithmName, alg.ComputeHash(data));
+            }
+        }
+
+        /// <summary>
+        ///   Conversion of a <see cref="MultiHash"/> to and from JSON.
+        /// </summary>
+        /// <remarks>
+        ///   The JSON is just a single string value.
+        /// </remarks>
+        private class Json : JsonConverter
+        {
+            public override bool CanConvert(Type objectType) { return true; }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'objectType'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)